custom_components/ramses_cc/climate.py

```diff
@@ -35,26 +78,104 @@ def entity_factory(entity_class, broker, device):  # TODO: deprecate
 
     platform = entity_platform.async_get_current_platform()
 
+    # Controller services
+    platform.async_register_entity_service(
+        SERVICE_RESET_SYSTEM_MODE,
+        cv.make_entity_service_schema(),
+        "svc_reset_system_mode",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_SYSTEM_MODE,
+        cv.make_entity_service_schema(
+            vol.Any(
+                SCHEMA_SYSTEM_MODE, SCHEMA_SYSTEM_MODE_HOURS, SCHEMA_SYSTEM_MODE_DAYS
+            )
+        ),
+        "svc_reset_system_mode",
+    )
+
+    # Zone services
+    platform.async_register_entity_service(
+        SERVICE_GET_ZONE_SCHED,
+        cv.make_entity_service_schema(),
+        "svc_get_zone_schedule",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_ZONE_SCHED,
+        cv.make_entity_service_schema({vol.Required(CONF_SCHEDULE): cv.string}),
+        "svc_set_zone_schedule",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_ZONE_SCHED,
+        cv.make_entity_service_schema({vol.Required(CONF_SCHEDULE): cv.string}),
+        "svc_set_zone_schedule",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_PUT_ZONE_TEMP,
+        cv.make_entity_service_schema(
+            {
+                vol.Required(CONF_TEMPERATURE): vol.All(
+                    vol.Coerce(float), vol.Range(min=-20, max=99)
+                ),
+            }
+        ),
+        "svc_put_zone_temp",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_ZONE_CONFIG,
+        cv.make_entity_service_schema(
+            {
+                vol.Optional(CONF_MAX_TEMP, default=35): vol.All(
+                    cv.positive_float,
+                    vol.Range(min=21, max=35),
+                ),
+                vol.Optional(CONF_MIN_TEMP, default=5): vol.All(
+                    cv.positive_float,
+                    vol.Range(min=5, max=21),
+                ),
+                vol.Optional(CONF_LOCAL_OVERRIDE, default=True): cv.boolean,
+                vol.Optional(CONF_OPENWINDOW, default=True): cv.boolean,
+                vol.Optional(CONF_MULTIROOM, default=True): cv.boolean,
+            }
+        ),
+        "svc_set_zone_config",
+    )
```

Comment:
It could go either way really, but this appeared to be the more common approach in the core components and I liked the simple flexibility it offered to toggle/adjust service availability in future depending on config.